operator/src/test/java/oracle/kubernetes/operator/helpers/DomainValidationStepTest.java

```diff
@@ -361,11 +383,52 @@ public void whenServerDoesNotExistInDomain_logWarning() {
         stringContainsInOrder("Managed Server", "no-such-server", "does not exist"));
   }
 
+  private DomainConfigurator configureDomain(Domain domain) {
+    return DomainConfiguratorFactory.forDomain(domain);
+  }
+
+  @Test
+  public void whenMonitoringExporterPortConflictsWithAdminServerPort_logWarning() {
+    configureDomain(domain).withMonitoringExporterConfiguration("queries:\n").withMonitoringExporterPort(7001);
+    testSupport.addToPacket(DOMAIN_TOPOLOGY, domainConfig);
+
+    testSupport.runSteps(topologyValidationStep);
+
+    assertThat(logRecords, containsWarning(MessageKeys.MONITORING_EXPORTER_CONFLICT_SERVER));
+    assertThat(info.getValidationWarningsAsString(),
+        stringContainsInOrder(Integer.toString(7001), ADMIN_SERVER, Integer.toString(ADMIN_SERVER_PORT_NUM)));
+  }
+
+  @Test
+  public void whenMonitoringExporterPortConflictsWithManagedServerPort_logWarning() {
+    configureDomain(domain).withMonitoringExporterConfiguration("queries:\n").withMonitoringExporterPort(8001);
+    testSupport.addToPacket(DOMAIN_TOPOLOGY, domainConfig);
+
+    testSupport.runSteps(topologyValidationStep);
+
+    assertThat(logRecords, containsWarning(MessageKeys.MONITORING_EXPORTER_CONFLICT_SERVER));
+    assertThat(info.getValidationWarningsAsString(),
+        stringContainsInOrder(Integer.toString(8001), MANAGED_SERVER1, Integer.toString(MANAGED_SERVER1_PORT_NUM)));
+  }
+
+  @Test
+  public void whenMonitoringExporterPortConflictsWithClusterServerTemplatePort_logWarning() {
+    configureDomain(domain).withMonitoringExporterConfiguration("queries:\n").withMonitoringExporterPort(9001);
+    testSupport.addToPacket(DOMAIN_TOPOLOGY, domainConfig);
+
+    testSupport.runSteps(topologyValidationStep);
+
+    assertThat(logRecords, containsWarning(MessageKeys.MONITORING_EXPORTER_CONFLICT_DYNAMIC_CLUSTER));
+    assertThat(info.getValidationWarningsAsString(),
+        stringContainsInOrder(Integer.toString(9001), DYNAMIC_CLUSTER_NAME,
+            Integer.toString(SERVER_TEMPLATE_PORT_NUM)));
```

Comment:
Do the namespace events and domain resource status also contain validation warnings (in addition to the operator log)?
If not, then they should - perhaps in a separate pull.
If yes, then it'd be a nice-to-have if the unit test checked for them.

Reply:
@doxiao, do these kind of domain validation errors that will appear in the domain status also show up in the event stream?

Reply:
I recall that Anthony added DomainValidationError event. @alai8 Can you confirm?

Reply:
The domain validation errors  should also show up in as events.

Reply:
@tbarnes-us, I found that some of the other unit-tests have checks that the event is generated. I'll do the same for these new tests.